src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -1019,6 +1018,23 @@ cpdef object compute_network_structured(
                 #tuple of MC_Reach and reach_type
                 MC_Reach(segment_objects, array('l',upstream_ids))
                 )
+            
+    cdef np.ndarray fill_index_mask = np.ones_like(data_idx, dtype=bool)
+    cdef Py_ssize_t fill_index
+    cdef long upstream_tw_id
+    cdef dict tmp
+    cdef int idx
+    cdef float val
+    
+    for upstream_tw_id in upstream_results:
+        tmp = upstream_results[upstream_tw_id]
+        fill_index = tmp["position_index"]
+        fill_index_mask[fill_index] = False
+        for idx, val in enumerate(tmp["results"]):            
+            if idx == 0:
+                flowveldepth_nd[fill_index, idx+1, 0] = val
+            else:
+                flowveldepth_nd[fill_index, np.floor(idx/3).astype('int')+1, np.mod(idx,3)] = val
```

Comment:
I used np.floor and np.mod to properly translate index values

Reply:
Don't use numpy here. Use integer division (// as James suggested) and the mod operator (%).
Using numpy here will introduce a high amount of overhead. Numpy is an array library is very unsuited for scalar calculations.
Some timings:
```
%timeit np.floor(5/3).astype('int') + 1, np.mod(5, 3)
The slowest run took 18.88 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 5: 3.2 µs per loop

%timeit (5//3)+ 1, 5 % 3
100000000 loops, best of 5: 9.69 ns per loop
```